### ISSUE TITLE
spawndist value is not used from DB

### ISSUE THREAD

**Issue by Krejza9** [TrinityCore/TrinityCore#26499]
**Description:**

in creature template there is value [spawndist](https://trinitycore.atlassian.net/wiki/spaces/tc/pages/2130009/creature)
but you don't use this value in the code at all and instead default value
`The spawndist value tells The maximum distance that the creature may spawn from its spawn point. Also controls how far away the creature can walk from its spawn point if its MovementType = 1.`

**Expected behaviour:**

You can easy fix it by modifing code
https://github.com/TrinityCore/TrinityCore/blob/d6dfb6c54c3296461ec73ea5d969f7a5a704541b/src/server/game/Movement/MovementGenerators/RandomMovementGenerator.cpp#L93
to 
`float distance = frand(0.f, 1.f) * owner->GetWanderDistance();`

**Steps to reproduce the problem:**

1. spawn creature
2. set MovementType = 1
3. set spawndist to 1
4. creature always walk 5 yards

**Branch(es):**

both

**TC rev. hash/commit:** 

CHANGEME Copy the result of server debug command (if you need to run it from client get prat addon)

**Operating system:** 

ALL

**Comment by ratkosrb** [comment]
```
The spawndist value tells The maximum distance that the creature may spawn from its spawn point.
```
To my knowledge that has never been true. It's just maximum wander distance for random movement generator.

**Comment by Krejza9** [comment]
maximum wander distance for random movement generator doesn't work anyway

**Comment by offl** [comment]
There's no spawndist column anymore in 3.3.5

> **Steps to reproduce the problem:**
>
>1. spawn creature
>2. set MovementType = 1
>3. set spawndist to 1
>4. creature always walk 5 yards

I can't reproduce on f62e57e295094018f85dc78d8d034b0167b8becf
Maybe you mean creature_template.MovementType = 1?

**Comment by Krejza9** [comment]
@offl I've uploaded a video to give you a better idea.
You can see that the distance from the spawnpoint doesn't work

https://youtu.be/D8sLZ8nqjKs

**Comment by Krejza9** [comment]
Afer fix  (It is still necessary to adjust the set value to make it take effect without restarting the server)
If you set wanderdistance 1 (NPC move 1 yard from its spawn point)
https://youtu.be/ESaNgYNgLL0

I'm adding values from the creatures table
![image](https://user-images.githubusercontent.com/5585555/116969660-eb802000-acb6-11eb-991c-30fd9968938f.png)

**Comment by offl** [comment]
> spawn creature

I only now realized you're spawning it by commands, then steps to reproduce should be

.npc add 6
.npc set movetype 1
.npc set wanderdistance 1

**Comment by Krejza9** [comment]
technically the command does not matter at all, the point is that the wander_distance value stored in the DB is not used in the code at all whether you use the command or not

**Comment by offl** [comment]
> the wander_distance value stored in the DB is not used in the c++ code

It works every time if creature is stored in DB, no matter if value is 1 or 3 or 15

**Comment by Krejza9** [comment]
I proved in the video provided that it doesn't work.

**Comment by offl** [comment]
SELECT * FROM creature WHERE wander_distance=1
Checked one yesterday

**Comment by Krejza9** [comment]
I think we don't understand each other, I'm not talking about the value in the DB, but about the fact that if the NPC has any value set, it still walks the same distance from the spawn.

**Comment by offl** [comment]
Yes, I meant if value is 1, creature will walk inside 1 yard, if it's 8, it will walk inside 8 yards

**Comment by Krejza9** [comment]
If you claim that it works for you **without** my fix, then I wonder what the problem is that it doesn't work for me.

**Comment by offl** [comment]
What's your rev?

**Comment by Krejza9** [comment]
Current commit f62e57e295094018f85dc78d8d034b0167b8becf

**Comment by Jinnaix** [comment]
> technically the command does not matter at all, the point is that the wander_distance value stored in the DB is not used in the c++ code, whether you use the command or not

The npc command value seems to get ignored. Creature will always pick wander_distance value from Database instead.
Check a creature with a bigger value.
e.g.
.go c 148
wander_distance value is 15. It will randomly move around in this range.
And it will fully ignore any different value you feed him with .npc set wanderdistance X
